lnwallet/commitment.go

```diff
@@ -613,6 +613,42 @@ func CommitScriptAnchors(chanType channeldb.ChannelType,
 	return localAnchor, remoteAnchor, nil
 }
 
+// CommitAuxLeaves stores two potential auxiliary leaves for the remote and
+// local output that made be used to argument the final tapscript trees of the
+// commitment transaction.
+type CommitAuxLeaves struct {
```

Comment:
Thinking about the name `CommitAuxLeaves` here... The most important auxiliary data is of course the `AuxTapLeaf`. But won't we also eventually store extra data related to a commitment transaction? Such as asset balances? Would that data live within this struct or alongside it within the custom blob?

Reply:
Yep, in the latest commits we add new TLV blobs to: `ChannelCommitment`, `OpenChannel`,  `RevocationLog`, `HTLCEntry` (htlcs in the revocation log). 

For funding, we'll store the information that doesn't change in `OpenChannel`. Then for each state `ChannelCommitment` will store the information specific to that state. Then finally for the revocations, we'll also store that set of minimal information. 

So at a high level, the upper layer will take that blob (format needs to be known/defined), then use that to create a tap leaf at the apprpriate output.